### PR TITLE
Resource count endpoint

### PR DESCRIPTION
**Pull request by sakhin123**
# Summary
We added API endpoints to do a resource count, and return a key:value pair of resourceTypes and their counts. 
## New behavior
New endpoint exists that gets a key:value pair of resourceTypes and their counts. 
## Code changes

Code changes made to:
- database/dbOperations.js, server/server.js and new files created to handle the new endpoint
- controllers/resourcecount.controller.js
- new function getAllResourceCounts() that retrieves all resource counts from db, then formats and returns them as key:value pairs
- services/resourcecount.service.js
- test/services/resourcecount.service.test.js

# Testing guidance
- GET request: http://localhost:3000/4_0_1/resourceCount
- Run unit tests

### CODE REVIEW

**Review by sarahmcdougall — CHANGES_REQUESTED** [review]
Great work on this task! :) I tested the endpoint in Insomnia and received accurate counts of the resources in my database. Also ran all the unit tests and they pass.

I don't have any suggested changes regarding functionality - everything seems good there. Just have a few comments on descriptions/documentation for you to take a look at. I also think this branch needs a good ol' rebase - I saw some `meta.lastUpdated` code and I think it's from a previous PR that recently got merged.

File: src/database/dbOperations.js [hunk 1]
```diff
@@ -286,6 +286,18 @@ const getCurrentSuccessfulImportCount = async clientId => {
   return bulkStatus.successCount;
 };
 
+/**
+ * gets the number of documents in a specified collection
+ * @param {*} resourceType specifies the name of the collection to be counted
```
[the remark this answers is not in the document]
Comment (sakhin123) [reply src/database/dbOperations.js:291]: Changed

File: test/services/resourcecount.service.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const testPatient = require('../fixtures/fhir-resources/testPatient.json');
+const testPatient2 = require('../fixtures/fhir-resources/testPatient2.json');
+const { testSetup, cleanUpTest } = require('../populateTestData');
+const { client } = require('../../src/database/connection');
+const supportedResources = require('../../src/server/supportedResources');
+
+let server;
+
+describe('resource count', () => {
+  beforeAll(() => {
```
[the remark this answers is not in the document]
Comment (sakhin123) [reply test/services/resourcecount.service.test.js:16]: Changed

File: test/services/base.service.test.js [hunk 3]
```diff
@@ -3,12 +3,23 @@ const supertest = require('supertest');
...
 const { SINGLE_AGENT_PROVENANCE } = require('../fixtures/provenanceFixtures');
+const { db } = require('../../src/database/connection');
 
 const updatePatient = { resourceType: 'Patient', id: 'testPatient', name: 'anUpdate' };
+
+const UPDATE_PATIENT_2 = { resourceType: 'Patient', id: 'testPatient2', name: 'anUpdate' };
+const UPDATE_PATIENT_3 = {
+  resourceType: 'Patient',
+  id: 'testPatient3',
+  meta: { lastUpdated: '1900-01-01T00:00:00Z' },
```
[the remark this answers is not in the document]
Comment (sakhin123) [reply test/services/base.service.test.js:19]: 👵🏼

**Review by sarahmcdougall — APPROVED** [review]

**Pull request merged**